engine/src/main/java/org/pentaho/di/trans/steps/databasejoin/DatabaseJoin.java

```diff
@@ -193,17 +193,21 @@ public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws
    *
    *
    * */
-  public synchronized void stopRunning( StepMetaInterface smi, StepDataInterface sdi ) throws KettleException {
+  public void stopRunning( StepMetaInterface smi, StepDataInterface sdi ) throws KettleException {
     if ( this.isStopped() || sdi.isDisposed() ) {
       return;
     }
     meta = (DatabaseJoinMeta) smi;
     data = (DatabaseJoinData) sdi;
 
-    if ( data.db != null && data.db.getConnection() != null && !data.isCanceled ) {
-      data.db.cancelStatement( data.pstmt );
-      setStopped( true );
-      data.isCanceled = true;
+    if ( data.db != null ) {
```

Comment:
Could introduce a race condition as there is still a synchronized method in this class. See https://github.com/pentaho/pentaho-kettle/blob/b0b2b733880f4b8f5e7b2349d9786457d65b37e4/engine/src/main/java/org/pentaho/di/trans/steps/databasejoin/DatabaseJoin.java#L59

Suggest getting Luc Boudreau to approve this before it's committed. Also suggest running a test with dozens of transformations with DatabaseJoin steps running simultaneously on the server. JMeter is particularly suited to help validate that this commit doesn't introduce a race condition.